File: android/rctmapboxar/src/main/java/com/mapbox/react/ar/math/Matrix4.java
package com.mapbox.react.ar.math;

public class Matrix4 {
    private float[] elements;

    public Matrix4() {
        elements = new float[]{
                1.f, 0.f, 0.f, 0.f,
                0.f, 1.f, 0.f, 0.f,
                0.f, 0.f, 1.f, 0.f,
                0.f, 0.f, 0.f, 1.f
        };
    }

    public float get(int index) {
        return elements[index];
    }
}

File: ios/RNMapboxAR/Matrix4.m
//
//  Matrix4.m
//  RNLocateTest
//
//  Created by Dave Prukop on 4/26/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "Matrix4.h"

@interface Matrix4()

@property (nonatomic, strong) NSArray<NSNumber *> *elements;

@end

@implementation Matrix4

- (instancetype)init
{
  self = [super init];
  if (self) {
    self.elements = [NSArray arrayWithObjects:
                     [NSNumber numberWithFloat:1.f], [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:0.f],
                     [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:1.f], [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:0.f],
                     [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:0.f],[NSNumber numberWithFloat:1.f], [NSNumber numberWithFloat:0.f],
                     [NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:0.f],[NSNumber numberWithFloat:0.f], [NSNumber numberWithFloat:1.f],
                     nil];
  }
  return self;
}

- (float)get:(NSUInteger)index
{
  return [[self.elements objectAtIndex:index] floatValue];
}

@end
